### DISCUSSION TITLE
esp32/Arduino/14.esp32-arduino-spi

### DISCUSSION THREAD

**Discussion by giscus** [bigrich-luo/DocsGiscus#27] [General]
# esp32/Arduino/14.esp32-arduino-spi

https://docs.geeksman.com/esp32/Arduino/14.esp32-arduino-spi.html

**Comment by yuanyuan19** [comment]
写的真心不错，为什么没人看

**Reply by elTusk** [reply]
支持

**Reply by OpenNine** [reply]
包支持的

**Reply by OpenNine** [reply]
来了

**Comment by eriojiang** [comment]
SPI（Serial Peripheral Interface，串行外设接口）的四根线可以通俗地理解为以下角色，想象它们是在一个团队中协作完成数据传输任务的成员：

片选线 (CS/SS - Chip Select/Slave Select)：这位成员是队长，负责告诉其他的队员（即从设备）何时开始准备接收或发送数据。当队长点名（将信号置为低电平）时，被选中的队员就要注意听了，没被点名的就保持安静。这样可以确保一次只有一个从设备与主设备沟通，避免混乱。
串行时钟线 (SCK - Serial Clock)：它是团队的节拍器，拿着一个小鼓，敲击出稳定的节奏。所有的数据传输动作都是跟着这个鼓点进行的，确保数据在传输过程中的同步。每当鼓点响起（时钟脉冲的上升沿或下降沿），队员们就知道是时候交换数据的一部分了。
主输出/从输入线 (MOSI - Master Out Slave In)：这是一条单向车道，主要用于主设备向从设备发送指令或数据。想象主设备是一个老师，通过这条线给学生（从设备）讲课，学生专心听讲并接收信息。
主输入/从输出线 (MISO - Master In Slave Out)：与MOSI相对，这是一条反向的单向车道，用于从设备向主设备反馈信息或数据。好比学生听完课后，通过这条线向老师提交作业或回答问题。

**Comment by Tzlzl12** [comment]
想不明白，如果只是调库那些算是屁个教程啊，连这起码的api都不说一下，尤其是前面的那个i2c
